### ISSUE TITLE
Chromecast connection in UI is lost when switching pages

### ISSUE THREAD

**Issue by cvium** [jellyfin/jellyfin-web#35]
## Expectation
Sender (app, server dashboard etc.) should be aware of its connection to Chromecast even after page reload.

It should be able to reconnect to an on-going casting session.
## Reality
Refreshing the page or even going to the server dashboard causes the connection to be lost and cannot be regained by connecting again.

It still continues to stream, but the sender can no longer control it.
## Steps to reproduce
1. Connect to CC via Chrome
2. Start a stream and make sure remote controls show up in Chrome
3. Go to server dashboard
4. CC icon is no longer active and re-connecting to CC has no effect.
4a. **NOTE**: Chrome tab is still connected to CC.

## Comments

It might be related to https://developers.google.com/cast/docs/reference/chrome/chrome.cast#.AutoJoinPolicy

**Comment by cvium** [comment]
Hmm, this appears to be working ?? Can you test it @anthonylavado ?

**Comment by anthonylavado** [comment]
Will test again

**Comment by anthonylavado** [comment]
@cvium - Just tested, still happening to me with Chrome Version 71.0.3578.98 (Official Build) (64-bit), macOS 10.14.2.

**Comment by JustSch** [comment]
this issues still occurs on win 10 1903 Chrome Version 80.0.3987.87 (Official Build) (64-bit)

**Comment by hamburglar2160** [comment]
This should be fixed with the new CAF receiver.

**Comment by JustSch** [comment]
is that part of the nightly release or has it not been pushed to chromecasts themselves yet? I tested it just now and still lost chromecast controls when going to dashboard jellyfin 10.5.0 Chrome version  83.0.4103.61 (Official Build) (64-bit) Win 10 1903

**Comment by hamburglar2160** [comment]
It's not available quite yet, but should be in 10.6 under the form of the 'nightly' Chromecast client when that gets built.

**Comment by MrTimscampi** [comment]
I don't think the new receiver app will change anything for this particular issue.

I see two issues with our current ChromecastPlayer (the sender part of the app):

* It uses the [Base API](https://developers.google.com/cast/docs/reference/chrome/chrome.cast) instead of the [Framework API](https://developers.google.com/cast/docs/reference/chrome/cast.framework). The Base API is lower-level than the Framework API and Google recommends using this last one when building a Sender app.
* During the bootstrap process for the app, we never check for an existing Cast session. So the session still exists, but it's never picked up by the ChromecastPlayer.

The first point will require a rewrite of the ChromecastPlayer. The second point will require some changes to the web client itself, but I'm unsure of the scope of these changes at the moment.

**Comment by brianjmurrell** [comment]
I don't think it even takes a page refresh or switching away to anywhere.  I can start casting leaving the browser on the show being cast and come back and the casting connection is gone with no ability to control the casting stream.  Trying to re-connect to the casting device ends the stream being casted.

**Comment by brianjmurrell** [comment]
This, frankly, seems much worse on the most recent release, 10.7.5.

Honestly, it makes using Jellyfin on anything other than a computer or phone quite useless.  I'm not sure about anyone else, but the idea of being able to watch the 42 or so minutes of a 60 minute program without a stop, pause, rewind, etc. is quite impossible.  Even trying to restart a stream and getting it play to where you left off -- once you have managed to get it to stop playing on a Chromecast device -- itself is even quite frustrating.

Are most people really using JF on computers and phones without Chromecast?  This just feels like a significant regression in the general usability of JF.  :-(

**Comment by brianjmurrell** [comment]
Hrm.  So the remote control functionality seems to work much better from a Linux Chrome browser than it does from Chrome on an Android phone.  Having to drag my laptop out to the livingroom, rather than just using my phone is still pretty low usability.

**Comment by PrplHaz4** [comment]
> 
> Are most people really using JF on computers and phones without Chromecast? This just feels like a significant regression in the general usability of JF. :-(

FireTV and DLNA have been a pretty good 10ft experience on my end - others are using AppleTV and Kodi successfully...
